jwst/extract_1d/extract.py

```diff
@@ -2733,118 +2568,127 @@ def do_extract1d(input_model, ref_dict, smoothing_length=None,
     if apply_nod_offset:
         if exp_type in WFSS_EXPTYPES + ['NRS_FIXEDSLIT', 'NRS_MSASPEC']:
             apply_nod_offset = False
-            log.warning("Correcting for nod/dither offset is currently "
-                        "not supported for exp_type = %s, so "
-                        "apply_nod_offset will be set to False",
-                        input_temp.meta.exposure.type)
+            log.warning(
+                f"Correcting for nod/dither offset is currently not supported for exp_type = "
+                f"{input_temp.meta.exposure.type}, so apply_nod_offset will be set to False",
+            )
 
     # Handle inputs that contain one or more slit models
-    if (was_source_model or
-        isinstance(input_model, datamodels.MultiSlitModel)):
-
-        if was_source_model:
-
-            # SourceContainer has a single list of SlitModels
+    if was_source_model or isinstance(input_model, datamodels.MultiSlitModel):
+        if was_source_model:   # SourceContainer has a single list of SlitModels
             slits = input_model
 
-            # The subsequent work on data uses the individual SlitModels,
-            # but there are many places where meta attributes are retreived
-            # from input_model, so set this to allow that to work
+            # The subsequent work on data uses the individual SlitModels, but there are many places where meta
+            # attributes are retreived from input_model, so set this to allow that to work.
             input_model = input_model[0]
-
-        elif isinstance(input_model, datamodels.MultiSlitModel):
-
-            # A simple MultiSlitModel, not in a container
+        elif isinstance(input_model, datamodels.MultiSlitModel):  # A simple MultiSlitModel, not in a container
             slits = input_model.slits
 
-        # Loop over the slits in the input model
-        for slit in slits:
-            log.info('Working on slit %s', slit.name)
+        for slit in slits:  # Loop over the slits in the input model
+            log.info(f'Working on slit {slit.name}')
             prev_offset = OFFSET_NOT_ASSIGNED_YET
+
             if np.size(slit.data) <= 0:
-                log.info('No data for slit %s, skipping ...', slit.name)
+                log.info(f'No data for slit {slit.name}, skipping ...')
                 continue
+
             sp_order = get_spectral_order(slit)
+
             if sp_order == 0 and not prism_mode:
                 log.info("Spectral order 0 is a direct image, skipping ...")
                 continue
+
             source_type = slit.source_type
+
             if source_type != 'POINT':
                 apply_nod_offset = False
-                log.warning("SRCTYPE = '%s'; correcting for nod/dither "
-                            "offset will only be done for a point source, "
-                            "so apply_nod_offset will be set to False",
-                            source_type)
+                log.warning(
```

Comment:
I don't think this needs to be a warning. Working on extended sources is perfectly natural and hence there's nothing bad about it. So change this to `log.info` instead.